src/dmd/target.d

```diff
@@ -1245,6 +1245,13 @@ struct TargetC
             bitFieldStyle = BitFieldStyle.Gcc_Clang;
         else
             assert(0);
+        /*
+            MacOS Monterey (12) does not support C runtime destructors.
```

Comment:
Strictly, it's not that they don't support, they choose to do so using atexit instead of having a dedicated binary section.